packages/react-instantsearch-core/src/connectors/__tests__/connectHitsInsights.ts

```diff
@@ -0,0 +1,103 @@
+import connect from '../connectHitsInsights';
+
+jest.mock('../../core/createConnector', () => x => x);
+
+function setup() {
+  const insightsClient = jest.fn();
+
+  const createMultiIndexContext = () => ({
+    context: {
+      ais: {
+        mainTargetedIndex: 'theFirstIndex',
+      },
+      multiIndexContext: {
+        targetedIndex: 'theIndex',
+      },
+    },
+  });
+
+  const context = createMultiIndexContext();
+  const getProvidedProps = connect(insightsClient).getProvidedProps.bind(
+    context
+  );
+
+  const hit = {
+    objectID: 'objectID_42',
+    __position: 42,
+    __queryID: 'theQueryID',
+  };
+  const searchResults = { results: { theIndex: { index: 'theIndex' } } };
+  const props = getProvidedProps({ hit }, null, searchResults);
+  return { insightsClient, props };
+}
+
+describe('connectHitsInsights', () => {
+  it('should expose an `insights` property', () => {
+    const { props } = setup();
+    expect(props).toHaveProperty('insights');
+  });
+
+  it('exposed `insights` should be a function', () => {
+    const { props } = setup();
+    expect(typeof props.insights).toBe('function');
+  });
+
+  describe('when called with `clickedObjectIDsAfterSearch`', () => {
+    let insightsClient;
+    beforeEach(() => {
+      const { insightsClient: aa, props } = setup();
+      insightsClient = aa;
+      props.insights('clickedObjectIDsAfterSearch', {
+        eventName: 'Add to cart',
+      });
+    });
+
+    it('should forward call to insightsClient with the correct payload', () => {
+      expect(insightsClient).toHaveBeenCalledTimes(1);
+      const [method, payload] = insightsClient.mock.calls[0];
+      expect(method).toEqual('clickedObjectIDsAfterSearch');
+      expect(payload.eventName).toEqual('Add to cart');
+      expect(payload.objectIDs).toEqual(['objectID_42']);
+      expect(payload.positions).toEqual([42]);
+      expect(payload.queryID).toEqual('theQueryID');
+      expect(payload.index).toEqual('theIndex');
```

Comment:
Prefer `toEqual` to strictly assert the the attributes of the payload.

Reply:
fixed here 283c42ba9c4bced043e72e3b5d67c4d1b1d0e3a3